### PR TITLE
Update custom permission rule naming and files

### PR DESCRIPTION
**Pull request by stanley-nicholl**
## Hey, I just made a Pull Request!
There was confusion with regards to some of the naming convention with the custom permissions documentation so this looks to simplify that work. The last [PR](https://github.com/backstage/backstage/pull/21542) looked to make it more understandable, but it was felt the naming convention (e.g.`isInSystemRuleFactory`) was a bad example (and technically incorrect as this is a condition factory, not a rule factory).

High-level, we remove the dedicated folder and file for custom permission rules/ conditions and keep everything contained to the `permission.ts` plugin file. However, as it pertains to this specific example, we are verbose in the code documentation to ensure users viewing the docs can be sure where to use the rule (i.e. catalog plugin) and the condition (i.e. permission plugin).

### Documentation on creating the rule and the condition
<table width="100%">
  <th>Previous</th>
  <th>Updated</th>
  <tr>
    <td width="50%">
      Screenshot 2023-12-21 at 17 21 40 
    </td>
    <td width="50%">
       Screenshot 2023-12-21 at 17 21 21
    </td>
  </tr>
</table>
*note: naming of the rule and the condition have reverted to the original naming convention as is recommended
  
### Documentation on using the condition in the permission plugin policy
<table width="100%">
  <th>Previous</th>
  <th>Updated</th>
  <tr>
    <td width="50%">
      Screenshot 2023-12-21 at 17 27 44
    </td>
    <td width="50%">
       Screenshot 2023-12-21 at 17 27 33
    </td>
  </tr>
</table>

### Documentation on using the condition in the permission plugin policy
<table width="100%">
  <th>Previous</th>
  <th>Updated</th>
  <tr>
    <td width="50%">
       Screenshot 2023-12-21 at 17 29 43
    </td>
    <td width="50%">
       Screenshot 2023-12-21 at 17 29 32
    </td>
  </tr>
</table>

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/permissions/custom-rules.md [hunk 1]
```diff
@@ -40,43 +42,48 @@ export const isInSystemRule = createCatalogPermissionRule({
...
-  const builder = await CatalogBuilder.create(env);
-  builder.addPermissionRules(isInSystemRule);
-  ...
-  return router;
-}
-```
-
-The new rule is now ready for use in a permission policy!
+import type { Entity } from '@backstage/catalog-model';
+import { createCatalogPermissionRule } from '@backstage/plugin-catalog-backend/alpha';
```
Comment (awanlin) [inline docs/permissions/custom-rules.md:58]:
```suggestion
import { catalogConditions, createCatalogConditionalDecision, createCatalogPermissionRule } from '@backstage/plugin-catalog-backend/alpha';
```

These are needed here related to the adds below

File: docs/permissions/custom-rules.md [hunk 2]
```diff
@@ -8,9 +8,11 @@ For some use cases, you may want to define custom [rules](./concepts.md#resource
...
+Plugins should export a rule factory that provides type-safety that ensures compatibility with the plugin's backend. The catalog plugin exports `createCatalogPermissionRule` from `@backstage/plugin-catalog-backend/alpha` for this purpose. Note: the `/alpha` path segment is temporary until this API is marked as stable. For this example, we'll define the rule and create a condition in `packages/backend/src/plugins/permission.ts`.
+
+```typescript title="packages/backend/src/plugins/permission.ts"
+...
 
-```typescript title="packages/backend/src/customPermissionRules/isInSystem.ts"
 import type { Entity } from '@backstage/catalog-model';
 import { createCatalogPermissionRule } from '@backstage/plugin-catalog-backend/alpha';
 import { createConditionFactory } from '@backstage/plugin-permission-node';
```
Comment (awanlin) [inline docs/permissions/custom-rules.md:18]:
```suggestion
import { catalogConditions, createCatalogConditionalDecision, createCatalogPermissionRule } from '@backstage/plugin-catalog-backend/alpha';
```

These are needed later

File: docs/permissions/custom-rules.md [hunk 3]
```diff
@@ -111,7 +118,31 @@ class TestPermissionPolicy implements PermissionPolicy {
...
+## Provide the rule during plugin setup
+
+Now that we have a custom rule defined and added to our policy, we need provide it to the catalog plugin. This step is important because the catalog plugin will use the rule's `toQuery` and `apply` methods while evaluating conditional authorize results. There's no guarantee that the catalog and permission backends are running on the same server, so we must explicitly link the rule to ensure that it's available at runtime.
+
+The api for providing custom rules may differ between plugins, but there should typically be some integration point during the creation of the backend router. For the catalog, this integration point is exposed via `CatalogBuilder.addPermissionRules`.
+
+```typescript title="packages/backend/src/plugins/catalog.ts"
+import { CatalogBuilder } from '@backstage/plugin-catalog-backend';
+/* highlight-add-next-line */
+import { isInSystemRule } from './permissions';
```
Comment (awanlin) [inline docs/permissions/custom-rules.md:130]: Sadly we get a type error with this:

image

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks for the adjustments @stanley-nicholl, took the time to actually implement this and ran into some problems which I've left comments for.

**Review by awanlin — APPROVED** [review]
Awesome work @stanley-nicholl 🚀

**Review by benjdlambert — APPROVED** [review]

**Comment by benjdlambert** [comment]
Thanks @stanley-nicholl! :tada:

**Pull request merged**